src/runtime/particle-spec.ts

```diff
@@ -525,30 +528,31 @@ export class ParticleSpec {
       if (s.tags.length > 0) {
         tokens.push(s.tags.map(a => `#${a}`).join(' '));
       }
-      results.push(`${indent}${tokens.join(' ')}`);
+      builder.push(`${tokens.join(' ')}`);
       if (s.provideSlotConnections) {
         // Provided slots.
-        s.provideSlotConnections.forEach(p => slotToString(p, 'provides', indent+'  '));
+        s.provideSlotConnections.forEach(p => slotToString(p, 'provides', builder.withIndent()));
       }
     };
 
-    this.slotConnections.forEach(
-      s => slotToString(s, 'consumes', '  ')
-    );
+    this.slotConnections.forEach(s => slotToString(s, 'consumes', indentedBuilder));
+
     // Description
     if (this.pattern) {
-      results.push(`  description \`${this.pattern}\``);
-      this.handleConnectionMap.forEach(cs => {
-        if (cs.pattern) {
-          results.push(`    ${cs.name} \`${cs.pattern}\``);
-        }
+      indentedBuilder.push(`description \`${this.pattern}\``);
+      indentedBuilder.withIndent(indentedBuilder => {
+        this.handleConnectionMap.forEach(cs => {
+          if (cs.pattern) {
+            indentedBuilder.push(`${cs.name} \`${cs.pattern}\``);
+          }
+        });
       });
     }
-    return results.join('\n');
+    return builder.toString();
   }
 
-  toManifestString(): string {
-    return this.toString();
+  toString(): string {
```

Comment:
hmm, unrelated to this PR, is there a reason why we need both?

Reply:
I'm not sure why we have both